templates/http/modules.j2

```diff
@@ -350,5 +350,37 @@ http2_recv_buffer_size {{ http2['recv_buffer_size'] }};
 {% if http2['chunk_size'] is defined %}
 http2_chunk_size {{ http2['chunk_size'] }};
 {% endif %}
+{% endmacro %}
+
+{# NGINX HTTP v3 -- ngx_http_v3_module #}
+{% macro http3(http3) %}
+{% if http3['enabled'] is defined and http3['enabled'] is boolean %}
+http3 {{ http3['enabled'] | ternary('on', 'off') }};
+{% endif %}
+{% if http3['hq'] is defined and http3['hq'] is boolean %}
+http3_hq {{ http3['hq'] | ternary('on', 'off') }};
+{% endif %}
+{% if http3['max_concurrent_streams'] is defined and http3['max_concurrent_streams'] is number %}
+http3_max_concurrent_streams {{ http3['max_concurrent_streams'] }};
+{% endif %}
+{% if http3['stream_buffer_size'] is defined %}
+http3_stream_buffer_size {{ http3['stream_buffer_size'] }};
+{% endif %}
 
 {% endmacro %}
+
+{# NGINX QUIC -- ngix_http_v3_module #}
+{% macro quic(quic) %}
+{% if quic['active_connection_id_limit'] is defined and quic['active_connection_id_limit'] is number %}
+quic_active_connection_id_limit {{ quic['active_connection_id_limit'] }};
+{% endif %}
+{% if quic['gso'] is defined and quic['gso'] is boolean %}
+quic_gso {{ quic['gso'] | ternary('on','off') }};
+{% endif %}
+{% if quic['host_key'] is defined %}
+quic_host_key {{ quic['host_key'] }};
+{% endif %}
+{% if quic['retry'] is defined and quic['retry'] is boolean %}
+quic_retry {{ quic['retry'] | ternary('on','off') }};
+{% endif %}
+{% endmacro %}
```

Comment:
I would ideally have these two within the same macro block since they are both part of the same module. I don't know what the best way to do that would be, but maybe we could do something where the resulting dictionary would be:
```yaml
httpv3:
  http3:
    hq: false
    max_concurrent_streams: 100
  quic:
    bpf: false
    gso: false
```
or:
```yaml
http3:
  hq: false
  max_concurrent_streams: 100
  quic:
    bpf: false
    gso: false
```

Reply:
It feels like keeping quic and http3 configuration separate is a sensible thing to do. Even if they are listed on a single page in the manual.
In nginx, quic related code is in `event/quic` and http3 is in `http/v3`. So it's not quite the same module. It's just a convenient way of representing it for now. 

If anything other than http3 will start using quic - it will appear as a separate module with the same options and we'll reuse this template.
And the manual is written the way it is, because in C code this is not a problem:  quic is already a separate entity and if quic moves into a separate module no configuration will have to be changed.

Hope this makes sense.

Reply:
It does!